test/e2e/framework/service/jig.go

```diff
@@ -773,6 +773,11 @@ func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v
 	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
 	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
 	for _, internalAddr := range internalAddrs {
+		// If the node's internal address points to localhost, then we are not
+		// able to test the service reachability via that address
+		if internalAddr == "127.0.0.1" {
```

Comment:
Don't we need to consider local loopback address of IPv6 here?

Reply:
[IsLoopback](https://golang.org/src/net/ip.go?s=3230:3260#L108) should do it, but need to `net.ParseIP` on the internal address first

Reply:
I added a separate utility function which verifies both.